lib/inputstreamhelper/utils.py

```diff
@@ -73,143 +72,168 @@ def download_path(url):
     return os.path.join(temp_path(), filename)
 
 
-def _http_request(url, headers=None, time_out=10):
-    """Perform an HTTP request and return request"""
-    log(0, 'Request URL: {url}', url=url)
-
+def _http_request(url, headers=None, time_out=30):
+    """Make a robust HTTP request handling redirections."""
     try:
-        if headers:
-            request = Request(url, headers=headers)
-        else:
-            request = Request(url)
-        req = urlopen(request, timeout=time_out)
-        log(0, 'Response code: {code}', code=req.getcode())
-        if 400 <= req.getcode() < 600:
-            raise HTTPError('HTTP {} Error for url: {}'.format(req.getcode(), url), response=req)
+        response = urlopen(url, timeout=time_out)  # pylint: disable=consider-using-with:w
+        if response.status in [301, 302, 303, 307, 308]:  # Handle redirections
+            new_url = response.getheader('Location')
+            log(1, f"Redirecting to {new_url}")
+            return _http_request(new_url, time_out)
+        return response  # Return the response for streaming
     except (HTTPError, URLError) as err:
         log(2, 'Download failed with error {}'.format(err))
-        if yesno_dialog(localize(30004), '{line1}\n{line2}'.format(line1=localize(30063), line2=localize(30065))):  # Internet down, try again?
```

Comment:
Please leave this comment in, so one doesn't have to look up what message 30065 says.